selendroid-server/src/main/java/io/selendroid/server/action/ActionChain.java

```diff
@@ -18,10 +18,11 @@
   private int pauseTime = 0;
   private int index = 0;
 
-  public ActionChain(JSONObject actionChain) throws JSONException {
+  public ActionChain(JSONObject actionChain, int i) throws JSONException {
     this.inputDevice = actionChain.getString("inputDevice");
     this.actionChain = actionChain.getJSONArray("actions");
     this.context = new ActionContext();
+    this.context.setId(i);
```

Comment:
As a disclaimer, I haven't used UUID's before so I could be wrong about this. From what I've read, UUID's are 128 bits but the unique pointer id used by Android's MotionEvents is an int. I don't think UUID's can be used to provide a unique integer. As an alternative, a class variable might work. What are your thoughts?